### PR TITLE
[R-package] standardize naming of internal functions

### PR DESCRIPTION
**Pull request by jameslamb**
Proposes removing the remaining uses of internal functions in the R package using dot-delimited names, like `lgb.is.null.handle()`.

Benefits of this change:

* minimizes the risk of `"Mismatches for apparent methods not registered:"` CRAN-release-blocking issues like https://github.com/dmlc/xgboost/pull/9577 and #5987
* standardizes the look of the project's R code a bit further

### CODE REVIEW

File: R-package/R/utils.R [hunk 1]
```diff
@@ -19,7 +19,7 @@ lgb.is.null.handle <- function(x) {
   )
 }
 
-lgb.params2str <- function(params) {
+.lgb_params2str <- function(params) {
```
Comment (jmoralez) [inline R-package/R/utils.R:22]: Did this function preserve the `lgb` prefix because it's a single word?

Comment (jameslamb) [reply R-package/R/utils.R]: hmm I didn't really think about the fact that this was the only one left with `lgb_`.

Just pushed a commit removing that: https://github.com/microsoft/LightGBM/pull/6179/commits/eb0ae49449959f18895a5ccf69fb34900ca1c50c

Thanks for catching the inconsistency.

File: R-package/R/lgb.cv.R [hunk 2]
```diff
@@ -594,7 +594,7 @@ lgb.stratified.folds <- function(y, k) {
   return(out)
 }
 
-lgb.merge.cv.result <- function(msg, showsd) {
+.lgb_merge_cv_result <- function(msg, showsd) {
```
Comment (jmoralez) [inline R-package/R/lgb.cv.R:597]:
```suggestion
.merge_cv_result <- function(msg, showsd) {
```

File: R-package/R/lgb.cv.R [hunk 3]
```diff
@@ -528,7 +528,7 @@ generate.cv.folds <- function(nfold, nrows, stratified, label, group, params) {
 # It was borrowed from caret::createFolds and simplified
 # by always returning an unnamed list of fold indices.
 #' @importFrom stats quantile
-lgb.stratified.folds <- function(y, k) {
+.lgb_stratified_folds <- function(y, k) {
```
Comment (jmoralez) [inline R-package/R/lgb.cv.R:531]:
```suggestion
.stratified_folds <- function(y, k) {
```

File: R-package/R/lgb.cv.R [hunk 4]
```diff
@@ -476,7 +476,7 @@ generate.cv.folds <- function(nfold, nrows, stratified, label, group, params) {
 
       y <- label[rnd_idx]
       y <- as.factor(y)
-      folds <- lgb.stratified.folds(y = y, k = nfold)
+      folds <- .lgb_stratified_folds(y = y, k = nfold)
```
Comment (jmoralez) [inline R-package/R/lgb.cv.R:479]:
```suggestion
      folds <- .stratified_folds(y = y, k = nfold)
```

File: R-package/R/lgb.cv.R [hunk 5]
```diff
@@ -387,7 +387,7 @@ lgb.cv <- function(params = list()
     })
 
     # Prepare collection of evaluation results
-    merged_msg <- lgb.merge.cv.result(
+    merged_msg <- .lgb_merge_cv_result(
```
Comment (jmoralez) [inline R-package/R/lgb.cv.R:390]:
```suggestion
    merged_msg <- .merge_cv_result(
```

**Review by jmoralez — COMMENTED** [review]
Sorry, missed these ones

**Review by jmoralez — APPROVED** [review]

**Pull request merged**